Meeting response models

juimin opened the pull request:
Check on the response and meeting models.

Commit:
unfinished meeting and response models. Contains sequelize data format -- requires review

juimin committed:
Merge branch 'master' into meeting-response-models

Commit:
Revert "unfinished meeting and response models. Contains sequelize data format -- requires review"

This reverts commit f336d1da3a40009e61b0c7ac22128658a7a9187f.

Commit:
Fixed the meeting and response models association issue

Commit:
Merge branch 'meeting-response-models' of https://github.com/EZPick/EasyPick into meeting-response-models

Commit:
fix the location test file

Commit:
change the data types to json for the schedule and the location preferences

Commit:
this has the json datatypes for postgres

Commit:
fix typo in Datatype -> Datatypes

juimin committed:
Merge branch 'master' into meeting-response-models

Commit:
add the starting date for all meetings and the ending date for all meetings

zmbc commented on src/server/app/models/meeting.js:
```diff
@@ -1,14 +1,23 @@
+    longitude: { type: DataTypes.INTEGER, defaultValue: null, validate: { min: -180, max: 180 }}, // Location longitude
+    radius: { type: DataTypes.INTEGER, defaultValue: 1000, allowNull: false }, // Location Search Radius in meters
+    duration: { type: DataTypes.INTEGER, allowNull: false, defaultValue: 60 },
+    participants: { type: DataTypes.JSON, defaultValue: "", allowNull: false },
```
What does `participants` mean? Isn't that just the responses?

zmbc commented on src/server/app/models/meeting.js:
```diff
@@ -1,14 +1,23 @@
+    closeoutDate: { type: DataTypes.DATE, allowNull: false},
+    closeoutTime: { type: DataTypes.TIME, allowNull: false},
+    meetingsStartDate: { type: DataTypes.TIME, allowNull: false},
+    meetingsEndDate: { type: DataTypes.TIME, allowNull: false},
```
What are `meetingsStartDate` and `meetingsEndDate`?

zmbc commented on src/server/app/models/meeting.js:
```diff
@@ -1,14 +1,23 @@
   }, {
     classMethods: {
-      associate: function (models) {
+        associate: function (models) {
```
unindent

zmbc commented on src/server/app/models/response.js:
```diff
@@ -1,10 +1,10 @@
-    locationPreferences: DataTypes.TEXT // Stored as JSON?
+    name: { type: DataTypes.STRING, allowNull: false }, // Responder Name
+    email: { type: DataTypes.STRING, allowNull: false, }, // Responder Email
+    schedule: { type: DataTypes.JSON, allowNull: false }, // Stored as JSON?
```
Remove `Stored as JSON?` comments.

zmbc commented on src/server/test/app/models/meeting.js:
```diff
@@ -2,6 +2,7 @@
 
 var expect = require('chai').expect;
 
+
```
Rm this change.

juimin committed:
Update meeting 

Participants changed to 'invited' and the meeting start and end dates removed

juimin committed:
Update meeting

Fix the indentation around classMethods & associate

juimin committed:
Updated Response.js

Removed the unneeded comments

zmbc commented on src/server/app/models/meeting.js:
```diff
@@ -1,14 +1,21 @@
-    generalLocation: DataTypes.STRING,
-    duration: DataTypes.INTEGER
+    title: { type: DataTypes.STRING, allowNull: false },
+    closeoutDate: { type: DataTypes.DATE, allowNull: false},
```
It says [here](http://docs.sequelizejs.com/variable/index.html#static-variable-DataTypes) that DataTypes.DATE is a datetime, so I don't think we need the DataTypes.TIME as well. I'd rename it back to `closeoutTime`.

zmbc commented on src/server/app/models/meeting.js:
```diff
@@ -1,14 +1,21 @@
+    radius: { type: DataTypes.INTEGER, defaultValue: 1000, allowNull: false }, // Location Search Radius in meters
+    duration: { type: DataTypes.INTEGER, allowNull: false, defaultValue: 60 },
+    invited: { type: DataTypes.JSON, defaultValue: "", allowNull: false },
+    creator: { type: DataTypes.STRING, defaultValue: "", allowNull: false } // Default value is one hour -- units in minutes
```
This comment should be on line 12, I'm guessing.

juimin committed:
Rm meeting js line

Empty line removed

juimin committed:
Meeting model - closeout Date + Time

DATE should be stored as a DATETIME so we don't need to have them separate

juimin committed:
Updated meeting model

The comment pertaining to meeting duration moved to the correct line

zmbc commented on src/server/app/models/meeting.js:
```diff
@@ -1,14 +1,20 @@
+    title: { type: DataTypes.STRING, allowNull: false },
+    closeoutTime: { type: DataTypes.DATE, allowNull: false},
+    generalLocation: { type: DataTypes.STRING, defaultValue: "", allowNull: false }, //  general location
+    latitude: { type: DataTypes.INTEGER, defaultValue: null, validate: { min: -90, max: 90 }}, // Location latitude
```
I think these should be `DOUBLE`s, not `INTEGER`s.

zmbc replied:
Wait a second, why are there longitude and latitudes on a meeting? These should be in the decision.

juimin committed:
update meeting lat long

The latitude and longitude are for the initial general location selection. Search radius is also included.

juimin committed:
update meeting js.

Clearer naming on the location latitude and longitude. Remove general meeting location string
